### DISCUSSION TITLE
Custom success messages from server action?

### DISCUSSION THREAD

**Discussion by bdrtsky** [edmundhung/conform#816] [Q&A]
How to return custom success message with submission.reply()?

**Comment by edmundhung** [comment]
You are free to change the shape of your action result

```tsx
return {
  result: submission.reply(),
  message: 'success',
}
```

As long as you pass the submission result back to the useForm hook

```tsx
const actionData = useActionData()
const [form, fields] = useForm({
  result: actionData?.result
})
```

**Reply by liamdawson** [reply]
I think it would be great to include this as an example in the docs, as this idea never occurred to me.

**Reply by subproject22** [reply]
@edmundhung This works, but how to make it work with typescript? it expects SubmissionResult[] from action
